class-zoninator-rest-api-controller.php

```diff
@@ -0,0 +1,575 @@
+                'args'                => $this->_get_zone_post_rest_route_params()
+            )
+        ) );
+
+        register_rest_route( $full_namespace, self::ZONE_ITEM_POSTS_POST_REGEX, array(
+            'methods'             => WP_REST_Server::DELETABLE,
+            'callback'            => array( $this, 'delete_item' ),
+            'permission_callback' => array( $this, 'delete_item_permissions_check' ),
+            'args'                => $this->_get_zone_post_rest_route_params()
+        ));
+
+        register_rest_route( $full_namespace, self::ZONE_ITEM_POSTS_URL_REGEX . '/order', array(
+            'methods'             => 'PUT',
+            'callback'            => array( $this, 'reorder_posts' ),
+            'permission_callback' => array( $this, 'update_item_permissions_check' ),
+            'args'                => $this->_get_zone_id_param()
+        ));
+
+        register_rest_route( $full_namespace, self::ZONE_ITEM_URL_REGEX . '/lock', array(
+            'methods'             => 'PUT',
+            'callback'            => array($this, 'zone_update_lock'),
+            'permission_callback' => array( $this, 'update_item_permissions_check' ),
+            'args'                => $this->_get_zone_id_param()
+        ));
+
+        register_rest_route( $full_namespace, '/posts/search', array(
+            'methods'  => WP_REST_Server::READABLE,
+            'callback' => array( $this, 'search_posts' ),
+            'args'     => $this->_params_for_search_posts()
+        ));
+
+        register_rest_route( $full_namespace, '/posts/recent', array(
+            'methods'  => WP_REST_Server::READABLE,
+            'callback' => array($this, 'get_recent_posts'),
+            'args'                => $this->_params_for_get_recent_posts()
+        ));
+    }
+
+    /**
+     * Create one item from the collection.
+     *
+     * @param WP_REST_Request $request Full data about the request.
+     * @return WP_Error|WP_REST_Response
+     */
+    function create_item( $request ) {
+        $zone_id = $this->_get_param( $request, 'zone_id', 0, 'absint' );
+        $post_id = $this->_get_param( $request, 'post_id', 0, 'absint' );
+
+        $post = get_post( $post_id );
+
+        if ( ! $post ) {
+            return $this->_bad_request(self::INVALID_POST_ID, __( 'invalid post id', self::ZONINATOR_NAMESPACE ));
+        }
+
+        $zone = $this->_zone_gateway->get_zone( $zone_id );
+
+        if ( ! $zone ) {
+            return $this->_bad_request(self::INVALID_ZONE_ID, __( 'invalid zone id', self::ZONINATOR_NAMESPACE ));
+        }
+
+        $result = $this->_zone_gateway->add_zone_posts( $zone_id, $post, true );
+
+        if ( is_wp_error( $result ) ) {
+            $status = 500;
+            $content = $result->get_error_message();
+        } else {
+            ob_start();
+            $this->_renderer->admin_page_zone_post( $post, $zone );
```

Comment:
Can we move the rendering down to the JS layer instead? Feels like the API should just return data and let clients handle the display aspect.